src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -497,39 +486,23 @@ iceberg_file_committer::commit_topic_files_to_catalog(
         auto dlq_table_id = table_id_provider::dlq_table_id(topic);
         auto dlq_table_res = co_await catalog_.load_table(dlq_table_id);
         if (dlq_table_res.has_error()) {
-            switch (dlq_table_res.error()) {
-            case iceberg::catalog::errc::not_found:
-                vlog(
-                  datalake_log.debug,
-                  "DLQ table {} not found for committing from topic {}",
-                  dlq_table_id,
-                  topic);
-                // We ignore the DLQ table not found error and continue under
-                // the assumption that there are will be no DLQ files to commit.
-                // If such a file is found we'll return an error. Ideally we
```

Comment:
To be clear, this comment was inaccurate before, right? i.e. we would hit a similar assertion (_not_ return an error) to the one described in the commit message iff there are pending dlq files?

Would there be any value to letting DLQ side fail lazily, as described here?

Reply:
Yea this was an oversight when the behavior changed here https://github.com/redpanda-data/redpanda/commit/510a9aa16163a56027fe6eba5324491e719cf267

>Would there be any value to letting DLQ side fail lazily, as described here?

I don't think so, we'll only load the DLQ table when we know there are DLQ files to commit. Letting the DLQ side slide means dropping some files on the floor, if i understand your question

Reply:
> only load tables with pending state

ya, makes sense. missed that, thanks